### PR TITLE
Add utility methods for job reconciliation

### PR DESCRIPTION
**Pull request by terrytangyuan**
This PR adds utility methods for job reconciliation (part of https://github.com/kubeflow/common/issues/7). Note that part of the code is commented out for now until service reconciliation logic is in place.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/common/24)

### CODE REVIEW

**Comment by terrytangyuan** [comment]
@richardsliu Coverage/coveralls calculation does not seem correct (it says `job_controller` folder's coverage decreased but I see no file under this directory seems to have decreased coverage). Would we still be able to merge this PR (once approved) while coveralls check failed?

File: job_controller/job_test.go [hunk 1]
```diff
@@ -83,6 +84,83 @@ func TestDeletePodsAndServices(T *testing.T) {
...
+
+func TestPastActiveDeadline(T *testing.T) {
+	type testCase struct {
+		activeDeadlineSeconds int64
+		expectedResult        bool
+	}
+
+	var testcase = []testCase{
+		{
+			int64(0),
```
Comment (richardsliu) [inline job_controller/job_test.go]: Same here

File: job_controller/job_test.go [hunk 2]
```diff
@@ -83,6 +84,83 @@ func TestDeletePodsAndServices(T *testing.T) {
...
 
+func TestPastBackoffLimit(T *testing.T) {
+	type testCase struct {
+		backOffLimit   int32
+		expectedResult bool
+	}
+
+	var testcase = []testCase{
+		{
+			int32(0),
```
Comment (richardsliu) [inline job_controller/job_test.go]: Comment what these parameters are for.

**Comment by terrytangyuan** [comment]
Thanks @richardsliu. Just updated so they are more readable now.

**Comment by gaocegege** [comment]
LGTM Is there anyone work on service reconciliation logic?

**Comment by terrytangyuan** [comment]
@gaocegege Yes someone from our team is putting together a PR (should be out soon).

**Pull request merged**